### PR TITLE
1 home page continued

### PR DESCRIPTION
**Pull request by Willo0ow**
- styles refactor
- added vue-router and axios
- use monte api
- filter screenings by date and/or movie 
- "contact us" section
- responsiveness (commit - add reactivity - I don't know where was my head when I wrote that message)

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -2,14 +2,26 @@
...
   </header>
-
   <main>
-    <HomePage />
+    <router-view></router-view>
   </main>
 </template>
-<script setup>
-import HomePage from "./pages/HomePage.vue";
+<script>
```
Comment (ghost) [inline src/App.vue:9]: I'm just curious, it's ok to do it both ways - why did you change `script setup` into options API?

File: src/App.vue [hunk 2]
```diff
@@ -2,14 +2,26 @@
...
 import NavBar from "./components/common/NavBar.vue";
+import { ref, provide } from "vue";
+export default {
+  components: { NavBar },
+  setup() {
+    const windowWidth = ref(window.innerWidth);
+    window.addEventListener("resize", () => {
```
Comment (ghost) [inline src/App.vue]: I saw you are injecting this value to check if it fits to some breakpoint. In this case you may consider to use just CSS to handle it 🤔 It's always good practice to use CSS instead of JS if it's possible and your case is good example for it :) in `MovieRow` component you can just make something like this: `Soon<span class="desktop-only"> in the Cinema</span>` and style the class.

Another thing (it's just for knowledge sharing since JS solution is not needed here) is that good practice is to add listeners when component is mounted and remove them when unmounted. So you could pass the adding listener in `onMounted` function and remove it inside `beforeUnmount` 👌

File: src/assets/styles/global/_breakpoints.scss [hunk 3]
```diff
@@ -0,0 +1,40 @@
+@mixin xl {
```
Comment (ghost) [inline src/assets/styles/global/_breakpoints.scss]: what you think to compose your breakpoints with `mobile first` rule?
in this approach initial styles (without media query) are valid from the smallest screen size. Then next breakpoint would be just `(min-width: SOME_WIDTH)` and all styles are valid for screens from `SOME_WIDTH` and above until next breakpoint is needed 👌 Advantage is that you can reuse styles for bigger screens without any action and change them only if they are different on some screen size. It's also cool because it's simple to read and understand because of this cascading way of building breakpoints 💪 

But it's just a proposition, I know it would take a lot of time to refactor as you already have a lot of styles in the app so it's up to you 👌

File: src/api/useMoviesApi.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import axios from "../axios";
+import { MOVIES } from "./endpoints.js";
+
+function useMoviesApi() {
+  async function retrieveMovies() {
+    try {
+      const res = await axios.get(`/${MOVIES}`);
```
Comment (ghost) [inline src/api/useMoviesApi.js:7]: `async/await` like a boss 💪

File: src/api/useSeancesApi.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import axios from "../axios";
+import { SEANCES } from "./endpoints.js";
+
+function useSeancesApi() {
+  async function retrieveDateSeances(date) {
```
Comment (ghost) [inline src/api/useSeancesApi.js]: BE made `seances` as one endpoint with possibility to add query params. What do you think to go with the convention and make just one function with optional params? It might looks like this:
```
async function retrieveSeances({ date, movieId } = {}) {
  try {
    const res = await axios.get(`/${SEANCES}`, {
      params: {
        data,
        movie_id: movieId,
      }
    );

    return res.data;
  }
  catch {...}
}
```

Then you can pass params in the object like `retrieveSeances({ date: SOME_DATE })` and it will also work like classic filters  👌

File: src/assets/styles/global/_mixins.scss [hunk 6]
```diff
@@ -0,0 +1,12 @@
+@mixin flex($direction: false, $justify: false, $align: false) {
```
Comment (ghost) [inline src/assets/styles/global/_mixins.scss:1]: 😎

File: src/assets/styles/global/_typography.scss [hunk 7]
```diff
@@ -0,0 +1,30 @@
+@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap');
+@import url('https://fonts.googleapis.com/css2?family=Roboto+Mono:wght@400;500;600;700&display=swap');
+@import url('https://fonts.googleapis.com/css2?family=Eczar:wght@400;500;600;700;800&display=swap');
+
+
+@mixin font($font, $family, $size, $weight, $lh, $color) {
```
Comment (ghost) [inline src/assets/styles/global/_typography.scss:6]: I like this abstraction and the usage of it below 😄

File: src/components/HomePage/ContactUs.vue [hunk 8]
```diff
@@ -1,9 +1,36 @@
 <template>
-  <div>{{ testData }}</div>
+  <div class="center-self contact-container">
+    <SectionTitle size="64" class="mb-48" style="text-align: center">
```
Comment (ghost) [inline src/components/HomePage/ContactUs.vue]: It's always a good practice to style elements with a `class` instead of `styles` 👌  it is easier to maintain this way. `style` rules are more important than these from `class` so it's harder to keep cascading behavior of css. Then only `!important` can help 🙀 but we don't want it 🙅

File: src/components/HomePage/MovieRow.vue [hunk 9]
```diff
@@ -1,57 +1,98 @@
...
-      },
-      {
-        id: 1,
-        title: "Terminator 2",
-        length: "1h 30min",
-        image: "terminator2.svg",
-        tags: ["Action"],
-      },
-    ];
+    const movieStore = useMovieStore();
```
Comment (ghost) [inline src/components/HomePage/MovieRow.vue]: Maybe `const { movies } = useMovieStore();` could be enough here? 🤔 but I may be wrong cause I never use Pinia 😅

File: src/components/HomePage/MovieRow.vue [hunk 10]
```diff
@@ -1,57 +1,98 @@
...
+    const { movies } = storeToRefs(movieStore);
+
+    const windowWidth = inject("windowWidth");
+    const soonText = computed(() =>
+      windowWidth.value > 480 ? "Soon in the Cinema" : "Soon"
+    );
+
+    const reducedMovies = computed(() =>
+      movies.value.reduce((soonMovies, movie) => {
```
Comment (ghost) [inline src/components/HomePage/MovieRow.vue]: maybe `movies.value.slice()` would be simpler here 😄 Reduce seems kinda overcomplicated for what is needed

File: src/components/HomePage/ContactUs.vue [hunk 11]
```diff
@@ -1,9 +1,36 @@
 <template>
-  <div>{{ testData }}</div>
+  <div class="center-self contact-container">
```
Comment (ghost) [inline src/components/HomePage/ContactUs.vue]: I saw in previous components you followed BEM convention but now you left this approach. It's good to keep it cause it's a nice way to scope styles and if whole project is made with one convention it's easier to maintain 👷

File: src/components/HomePage/ContactUs.vue [hunk 12]
```diff
@@ -1,9 +1,36 @@
...
 <script>
+import SectionTitle from "@components/common/SectionTitle.vue";
+import LocationIcon from "@assets/images/icons/locationIcon.svg?component";
+import PhoneIcon from "@assets/images/icons/phoneIcon.svg?component";
+import EmailIcon from "@assets/images/icons/emailIcon.svg?component";
 export default {
-  components: {},
+  components: { SectionTitle, LocationIcon, PhoneIcon, EmailIcon },
   setup() {
     const testData = "Contact Us";
```
Comment (ghost) [inline src/components/HomePage/ContactUs.vue]: it's probably unused now 👌 you can kick it off 🦿

File: src/components/HomePage/Screenings.vue [hunk 13]
```diff
@@ -1,29 +1,127 @@
...
+      <SectionTitle size="64">Screenings:</SectionTitle>
+      <SectionTitle size="64" color="bombay" class="mb-32">{{
+        dateString
+      }}</SectionTitle>
+    </div>
+    <div class="filters-container mb-64">
+      <div class="days-container">
+        <SelectDayFilter
+          @updateCurrentDate="updateCurrentDate"
+        ></SelectDayFilter>
```
Comment (ghost) [inline src/components/HomePage/Screenings.vue]: It's ok here but as a tip - for Components without slots you can also use self closing tags like `<SelectDayFilter :someProp="someValue" />` 😸

File: src/components/HomePage/Screenings.vue [hunk 14]
```diff
@@ -1,29 +1,127 @@
...
+          .replace(/\./g, "/")}`
+    );
 
     const movies = useTestMovies();
+    const movieStore = useMovieStore();
+    const seanceStore = useSeanceStore();
+
+    onMounted(async () => {
+      await movieStore.getMovies();
+      seanceStore.getDateSeances("21-08-2022");
```
Comment (ghost) [inline src/components/HomePage/Screenings.vue]: Here is mocked date so maybe it should be `currentDate.value` instead 🤔

File: src/components/HomePage/SelectDayFilter.vue [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+.filter-days {
+  @include flex();
+  overflow-x: scroll;
+}
+.filter-days button {
+  align-self: center;
+}
+.hide-day {
+  @include md {
+    &:nth-last-child(3) {
```
Comment (ghost) [inline src/components/HomePage/SelectDayFilter.vue]: You could make it shorter with `&:nth-last-child(3), &:nth-last-child(2) {` as they should have the same styling 😎

File: src/components/HomePage/Screenings.vue [hunk 16]
```diff
@@ -1,29 +1,127 @@
...
+    }
+
+    const dateString = computed(
+      () =>
+        `${weekdays[currentDate.value.getDay()].fullName} ${currentDate.value
+          .toLocaleDateString()
+          .replace(/\./g, "/")}`
+    );
 
     const movies = useTestMovies();
```
Comment (ghost) [inline src/components/HomePage/Screenings.vue]: this one is probably not used anymore 🚜

File: src/components/common/CustomSelect.vue [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+</script>
+<style lang="scss">
+.custom-select {
+  font-family: "Roboto", sans-serif !important;
+  font-style: normal !important;
+  font-weight: 400 !important;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+.custom-select .vs__dropdown-toggle {
```
Comment (ghost) [inline src/components/common/CustomSelect.vue]: you can use SCSS magic here with `& .vs__dropdown-toggle {` as everywhere else 🪄

File: src/components/common/MovieCard.vue [hunk 18]
```diff
@@ -41,11 +33,22 @@ export default {
...
+  @include flex(column, space-between);
+  @include max-sm {
+    padding: 24px;
+  }
 }
 .card__image {
   margin: 18px 0;
 }
+img {
+  // width: 341px;
```
Comment (ghost) [inline src/components/common/MovieCard.vue]: just a left comment 👮

File: src/components/common/MovieCardDetailed.vue [hunk 19]
```diff
@@ -43,13 +60,16 @@ export default {
       import.meta.url
     ).href;
 
+    const selectedScreening = ref("");
```
Comment (ghost) [inline src/components/common/MovieCardDetailed.vue:63]: I can't see cause I have no BE app cloned but how this selected screens look initially? is some of them selected or they are collapsed and wait for check one?

File: src/components/common/MovieList.vue [hunk 20]
```diff
@@ -3,7 +3,7 @@
     <MovieCardDetailed
       class="mb-40"
       v-for="movie in movies"
-      :key="movie.title"
+      :key="movie.id"
```
Comment (ghost) [inline src/components/common/MovieList.vue:6]: 💪

File: src/components/icons/CalendarIcon.vue [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+    />
+  </svg>
+</template>
+<script>
+import { ref, computed } from "vue";
+export default {
+  props: { color: { type: String, default: "#343541" } },
+  setup(props) {
+    const isHovered = ref(false);
+    function toggleHover() {
```
Comment (ghost) [inline src/components/icons/CalendarIcon.vue]: with a rule to always try to do something with CSS if possible you can make this icon hoverable just with CSS in the place of usage 😄

File: src/stores/movies.js [hunk 22]
```diff
@@ -0,0 +1,28 @@
...
+  actions: {
+    async getMovies() {
+      const res = await retrieveMovies();
+      this.movies = res.map((movie) => {
+        const { length, genre, poster_url } = movie;
+        const hours = Math.floor(length / 60);
+        const minutes = `0${length % 60}`.slice(-2);
+        const lengthInString = `${hours}h ${minutes}min`;
+        return {
+          ...movie,
```
Comment (ghost) [inline src/stores/movies.js:20]: I'm not sure if such behavior is wanted but if you would like to omit destructured properties from `movie` you can declare in 15 line something like `const { length, genre, poster_url, ...restMovie } = movie;` and then spread here this `restMovie` instead of `movie` 😄 but if `genre` and `poster_url` are expected (or just can exist because they don't break anything) to be in returned `movie` it's cool as it is 👌

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/App.vue (hunk 2 above)
Comment (Willo0ow) [reply src/App.vue]: Yeah, I'm more of JS person than CSS, and styles is the field I want to develop the most during this bootcamp. Love the idea of desktop-only class, I'll implement it.
When it comes to eventListeners - very valuable tip,  I don't have a habit of removing them and that's something that should change.

File: src/api/useSeancesApi.js (hunk 5 above)
Comment (Willo0ow) [reply src/api/useSeancesApi.js]: Agree, less code but still easy to interpret what the function does.

File: src/assets/styles/global/_breakpoints.scss (hunk 3 above)
Comment (Willo0ow) [reply src/assets/styles/global/_breakpoints.scss]: I'm not satisfied how I handled breakpoints, adjusting styles to screen size with my breakpoints is quite tiresome, your suggestion sounds like a better solution. I'll go with refactor, better do it now than later.

File: src/components/HomePage/MovieRow.vue (hunk 9 above)
Comment (Willo0ow) [reply src/components/HomePage/MovieRow.vue]: I guess it could look like this: `const { movies } = storeToRefs(useMovieStore());` since i need to watch changes in store

File: src/components/HomePage/Screenings.vue (hunk 14 above)
Comment (Willo0ow) [reply src/components/HomePage/Screenings.vue]: Yep, it's an artefact

File: src/components/common/MovieCardDetailed.vue (hunk 19 above)
Comment (Willo0ow) [reply src/components/common/MovieCardDetailed.vue:63]: Initial:
Zrzut ekranu 2022-08-26 o 09 39 37
Selected:
Zrzut ekranu 2022-08-26 o 09 41 03
I guess, in the future, clicking on seance btn will redirect to "make a reservation" page.

File: src/components/icons/CalendarIcon.vue (hunk 21 above)
Comment (Willo0ow) [reply src/components/icons/CalendarIcon.vue]: I realized that when working on another branch. I went even further, I defined styles for hover on icon from button component using :deep(), since I only want the icon to change its color on hover only when it's inside a button.

File: src/stores/movies.js (hunk 22 above)
Comment (Willo0ow) [reply src/stores/movies.js:20]: They don't break anything, at least for now. Some of them will actually be useful, e.g. genre.id in the future.

File: src/components/HomePage/ContactUs.vue (hunk 11 above)
Comment (Willo0ow) [reply src/components/HomePage/ContactUs.vue]: This is the first time that I tried to follow BEM, in some components it's easy to find names for classes according to the convention, but in other I found it difficult. I guess it takes practice. I'll fix it.

File: src/components/common/MovieCardDetailed.vue (hunk 19 above)
Comment (ghost) [reply src/components/common/MovieCardDetailed.vue:63]: got it 👌 looks good 🌞

File: src/components/icons/CalendarIcon.vue (hunk 21 above)
Comment (ghost) [reply src/components/icons/CalendarIcon.vue]: got it 🫡 
And I think there is another and probably the simplest solution for it 🤔 you can use the newest Vue magic, use `v-bind(VALUE)` directly in `style` block. you can get props with colors and handle these colors with css using `:hover` pseudoclass 👌 what you think about it? more info -> https://vuejs.org/api/sfc-css-features.html#v-bind-in-css

Comment (Willo0ow) [reply src/components/icons/CalendarIcon.vue]: I've read about it, it was tempting to use it in this case.
why I didn't?
Let's think about it. When do we want the icon to change its color? When hover on icon only? Or when the button, in which the icon is, is hovered? So is it actually a good idea to define :hover on icon?
This is how I tried to solve the problem: 
`      .custom-button:hover {
        background: #212229;
        color: white;
        & :deep(.icon *) {
          stroke: white;
        }
      }`

Comment (ghost) [reply src/components/icons/CalendarIcon.vue]: totally agree 👍 and this way you have access to SCSS variables for the icon

File: src/App.vue (hunk 1 above)
Comment (Willo0ow) [reply src/App.vue:9]: it's still composition api but setup is a function inside the script. Why the change - I find it easier to write the logic in setup function than in script setup. When I created the project with vite, App component was by default in script setup. And as long as I didn't have any logic inside I didn't mind.

File: src/components/HomePage/Screenings.vue (hunk 16 above)
Comment (Willo0ow) [reply src/components/HomePage/Screenings.vue]: I still use it to create nice 'Monday 22/08/2022' string from a simple date object.

Comment (ghost) [reply src/components/HomePage/Screenings.vue]: I mean just `const movies = useTestMovies();` :) probably GH attached the comment to last edited part of code 😅

File: src/components/common/CustomMenu.vue [hunk 23]
```diff
@@ -0,0 +1,47 @@
+<template>
+  <div class="menu">
+    <NavigationIcon class="menu__icon" v-s-dropdown-toggle:menu__options />
+    <SDropdown name="menu__options">
+      <ul>
+        <li class="menu__option" v-for="tab in tabs" :key="tab.label">
+          <a :href="tab.path">{{ tab.label }}</a>
+        </li>
+        <hr class="menu__divider" />
+        <li class="menu__option"><a href="/login">Login</a></li>
```
Comment (ghost) [inline src/components/common/CustomMenu.vue]: here should be `router-link` instead of `a` as it is routing inside the app 👌

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/components/common/CustomMenu.vue (hunk 23 above)
Comment (Willo0ow) [reply src/components/common/CustomMenu.vue]: achhhhh racja

**Review by ghost — APPROVED** [review]
